src/requestforattestation/RequestForAttestation.ts

```diff
@@ -88,56 +94,91 @@ export default class RequestForAttestation implements IRequestForAttestation {
   }
 
   /**
-   * [STATIC] Builds a new instance of [[RequestForAttestation]], from a complete set of requiered parameters.
+   * [STATIC] Builds a new instance of [[RequestForAttestation]], from a complete set of required parameters.
    *
-   * @param claimInput - An `IClaim` object the request for attestation is built for.
-   * @param identity - The Claimer's [Identity].
-   * @param legitimationsInput - Array of [AttestedClaim] objects of the Attester which the Claimer requests to include into the attestation as legitimations.
-   * @param delegationIdInput - The id of the DelegationNode of the Attester, which should be used in the attestation.
-   * @returns  A new [[RequestForAttestation]] object.
-   * @example ```javascript
-   * const requestForAttestation = RequestForAttestation.fromClaimAndIdentity(
-   *   claim,
-   *   alice
-   * );
-   * ```
+   * @param p - Parameter object.
+   * @param p.claim - An `IClaim` object the request for attestation is built for.
+   * @param p.identity - The Claimer's [Identity].
+   * @param p.legitimations - Array of [AttestedClaim] objects of the Attester which the Claimer requests to include into the attestation as legitimations.
+   * @param p.delegationId - The id of the DelegationNode of the Attester, which should be used in the attestation.
+   * @param p.privacyEnhanced -
+   * @param p.privacyEnhanced -
+   * @returns A new [[RequestForAttestation]] object.
    */
-  public static fromClaimAndIdentity(
-    claimInput: IClaim,
-    identity: Identity,
-    legitimationsInput: AttestedClaim[] = [],
-    delegationIdInput: IDelegationBaseNode['id'] | null = null
-  ): RequestForAttestation {
-    if (claimInput.owner !== identity.address) {
+  public static async fromClaimAndIdentity({
+    claim,
+    identity,
+    legitimations,
+    delegationId,
+    initiateAttestationMsg,
+    attesterPubKey,
+  }: {
+    claim: IClaim
```

Comment:
As discussed, have `claim` and `identity` as direct parameter and the optional as `Option` param

Reply:
fixed in 54fee04f0f30febd9b850c5522ee2100116475f7